src/function/cast/struct_cast.cpp

```diff
@@ -229,6 +229,120 @@ static bool StructToVarcharCast(Vector &source, Vector &result, idx_t count, Cas
 	return true;
 }
 
+unique_ptr<BoundCastData> StructToMapBoundCastData::BindStructToMapCast(BindCastInput &input, const LogicalType &source,
+                                                                        const LogicalType &target) {
+	if (StructType::IsUnnamed(source)) {
+		throw TypeMismatchException(input.query_location, source, target, "Cannot cast unnamed STRUCTs to MAP");
+	}
+	// Get a cast function for the keys (struct has varchar keys, map has single-type keys)
+	auto key_cast = input.GetCastFunction(LogicalType::VARCHAR, MapType::KeyType(target));
+	// Get cast function for the values (struct may have different value types per key, map has single-type values)
+	vector<BoundCastInfo> value_casts;
+	auto target_value_type = MapType::ValueType(target);
+	for (idx_t i = 0; i < StructType::GetChildCount(source); i++) {
+		auto value_cast = input.GetCastFunction(StructType::GetChildType(source, i), target_value_type);
+		value_casts.push_back(std::move(value_cast));
+	}
+	return make_uniq<StructToMapBoundCastData>(std::move(key_cast), std::move(value_casts));
+}
+
+unique_ptr<FunctionLocalState>
+StructToMapBoundCastData::InitStructToMapCastLocalState(CastLocalStateParameters &parameters) {
+	auto &cast_data = parameters.cast_data->Cast<StructToMapBoundCastData>();
+	auto result = make_uniq<StructToMapCastLocalState>();
+
+	if (cast_data.key_cast.init_local_state) {
+		CastLocalStateParameters child_params(parameters, cast_data.key_cast.cast_data);
+		result->key_state = cast_data.key_cast.init_local_state(child_params);
+	}
+
+	for (auto &entry : cast_data.value_casts) {
+		unique_ptr<FunctionLocalState> child_state;
+		if (entry.init_local_state) {
+			CastLocalStateParameters child_params(parameters, entry.cast_data);
+			child_state = entry.init_local_state(child_params);
+		}
+		result->value_states.push_back(std::move(child_state));
+	}
+	return std::move(result);
+}
+
+static bool StructToMapCast(Vector &source, Vector &result, idx_t count, CastParameters &parameters) {
+
+	if (source.GetVectorType() == VectorType::CONSTANT_VECTOR) {
+		// If the source is a constant vector the result may be as well
```

Comment:
I think we can set `count` to 1, if we're dealing with a constant vector?

Reply:
We're setting `total_count` to `count * field_count`, I'm saying we should probably first do `count = 1`
Since this is a constant vector, it physically only has a single row, no matter what the count is

Reply:
When you try to special case for a VectorType it adds a little complexity
It can be more optimized as a result, but it does add complexity

Most scalar functions (though this isn't a scalar function, it's a cast, but it's mostly the same thing) do this, at the very end:
```
	if (source.GetVectorType() == VectorType::CONSTANT_VECTOR) {
		result.SetVectorType(VectorType::CONSTANT_VECTOR);
	}
```
So it can treat the result as a FlatVector while populating it, only turning it into a constant at the end